doc/migrations/v0.45-v0.46.md

```diff
@@ -0,0 +1,119 @@
+# Migrating to libp2p@46 <!-- omit in toc -->
+
+A migration guide for refactoring your application code from libp2p v__ to v__.
+
+## Table of Contents <!-- omit in toc -->
+
+- [Graceful stream closing](#graceful-stream-closing)
+- [Stream/Connection stat properties](#streamconnection-stat-properties)
+- [Interface module consolidation](#interface-module-consolidation)
+
+## Graceful stream closing
+
+Streams can either be closed gracefully, where we wait for any unsent data to be sent, or aborted in which case any unsent data is discarded and a reset message is sent, notifying the remote of the abnormal termination.
+
+To close a stream gracefully we call the `.close` method (or `.closeRead`/`.closeWrite` for when we want half-closed streams). To abort a stream we call `.abort` and pass an error object.
+
+In previous versions the `.close` method was synchronous which meant it could not wait for existing data to be sent which made nodes behave unpredictably.
+
+From `0.46.x` the `.close`/`.closeRead`/`.closeWrite` methods on the Stream interface are now asynchronous.  `.abort` is a synchronous method that accepts an Error object.
+
+Similarly the Connection interface now has asynchronous `.close` and synchronous `.abort` methods.
+
+The `.reset` method has been removed from the Stream interface as it is only to be invoked internally by stream multiplexers when a remote stream reset has occurred.
+
+**Before**
+
+```js
```

Comment:
Going forward we should really utilize Typescript examples so that they can be compiled and thus reduce the probability of doc errors.

Reply:
Probably, though it'll involve a lot of boilerplate `import`s.

Reply:
True, but otherwise we lose the benefit of the type safety. Perhaps we can begin with https://github.com/libp2p/js-libp2p/issues/1777 as a precedent.